### PR TITLE
Update details on dashboard and `view` command implementation

### LINKED ISSUE #218: Add Dashboard into UI for DG?
image

@zhyuhan This one is the high level overview of the various components, but do we want to add one more for the dashboard here?

### PR DESCRIPTION
**Pull request by zhyuhan**
Fixes #218.

### CODE REVIEW

**Review by KamiliArsyad — APPROVED** [review]
LGTM

**Review by MagnificentCreature — APPROVED** [review]
lgtm I will force this in

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -173,19 +173,36 @@ The class also has getters and setters for the fields, as well as a `toString()`
...
+
+In order to achieve real-time updates, we will need to implement a mechanism to track changes to the client data, and trigger the dashboard to update accordingly. However, this leads to another problem, which is the performance of the dashboard.
+
+As the amount of client data increases, the calculation of statistics and follow-ups may increase significantly. Therefore, the dashboard should be updated, and the statistics and reminders data recalculated, only when the dashboard view is actually shown, and data has been modified.
+
+The `Dashboard` component already has an `isDashboardDirty` flag to indicate if the dashboard data needs to be recalculated. However, due to time constraint, it is currently always set to `true` and not being used to track data changes to the client data. In the future, we can track changes to this flag, e.g. using observers, and update the dashboard accordingly.
```
Comment (KamiliArsyad) [inline docs/DeveloperGuide.md:209]: Should we add a future thingy section instead here or isit alright to leave the description of how it should be in the future like this?

**Review by KamiliArsyad — APPROVED** [review]
little nit but lgtm

**Pull request merged**